### PR TITLE
Change multiopen commitment scheme to KZG

### PR DESCRIPTION
**Pull request by kilic**
This large PR replaces inner product argument commitments with KZG. Batch KZG commitment scheme implemented here closely follows section 3 of [plonk paper](https://eprint.iacr.org/2019/953.pdf).

### CODE REVIEW

File: src/poly/commitment.rs [hunk 1]
```diff
@@ -30,69 +23,72 @@ pub struct Params<C: CurveAffine> {
...
+    _marker: PhantomData<E>,
+}
 
+impl<E: Engine> Setup<E> {
+    /// Initializes parameters for the curve, given a random oracle to draw
+    /// points from.
+    pub fn new(k: u32, mut rng: impl RngCore) -> Params<E::G1Affine> {
+        // Largest root of unity exponent of the Engine is `2^E::Scalar::S`, so we can
+        // only support FFTs of polynomials below degree `2^E::Scalar::S`.
+        assert!(k < E::Scalar::S);
```
Comment (therealyingtong) [inline src/poly/commitment.rs:52]: Should this be `<=`?

**Review by therealyingtong — APPROVED** [review]
LGTM modulo a question.

**Review by han0110 — APPROVED** [review]
LGTM!

File: src/poly/commitment.rs [hunk 2]
```diff
@@ -30,69 +23,69 @@ pub struct Params<C: CurveAffine> {
     pub(crate) n: u64,
     pub(crate) g: Vec<C>,
     pub(crate) g_lagrange: Vec<C>,
-    pub(crate) h: C,
-    pub(crate) u: C,
+    pub(crate) additional_data: Vec<u8>,
```
[the remark this answers is not in the document]
Comment (kilic) [reply src/poly/commitment.rs:26]: Since this suggested approach requires so many changes I opened another PR for it. #3

**Pull request merged**